moveit_ros/planning_interface/robot_interface/src/wrap_python_robot_interface.cpp

```diff
@@ -34,114 +34,115 @@
 
 /* Author: Ioan Sucan */
 
+#ifndef PY_SSIZE_T_CLEAN
+#define PY_SSIZE_T_CLEAN
+#endif
+
+#include <moveit/robot_model/robot_model.h>
 #include <moveit/common_planning_interface_objects/common_objects.h>
 #include <moveit/robot_state/conversions.h>
 #include <moveit/py_bindings_tools/roscpp_initializer.h>
-#include <moveit/py_bindings_tools/py_conversions.h>
-#include <moveit/py_bindings_tools/serialize_msg.h>
-#include <moveit/py_bindings_tools/gil_releaser.h>
+#include <moveit/py_bindings_tools/ros_msg_typecasters.h>
 #include <moveit_msgs/RobotState.h>
 #include <visualization_msgs/MarkerArray.h>
 
 #include <stdexcept>
-#include <boost/python.hpp>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
 #include <Python.h>
 
 /** @cond IGNORE */
 
-namespace bp = boost::python;
-using moveit::py_bindings_tools::GILReleaser;
+namespace py = pybind11;
+namespace pi = moveit::planning_interface;
 
-namespace moveit
-{
-class RobotInterfacePython : protected py_bindings_tools::ROScppInitializer
+class RobotInterfacePython : protected moveit::py_bindings_tools::ROScppInitializer
 {
 public:
   RobotInterfacePython(const std::string& robot_description, const std::string& ns = "")
-    : py_bindings_tools::ROScppInitializer()
+    : moveit::py_bindings_tools::ROScppInitializer()
   {
-    robot_model_ = planning_interface::getSharedRobotModel(robot_description);
+    robot_model_ = pi::getSharedRobotModel(robot_description);
     if (!robot_model_)
       throw std::runtime_error("RobotInterfacePython: invalid robot model");
-    current_state_monitor_ =
-        planning_interface::getSharedStateMonitor(robot_model_, planning_interface::getSharedTF(), ns);
+    current_state_monitor_ = pi::getSharedStateMonitor(robot_model_, pi::getSharedTF(), ns);
   }
 
   const char* getRobotName() const
   {
     return robot_model_->getName().c_str();
   }
 
-  bp::list getActiveJointNames() const
+  std::vector<std::string> getActiveJointNames() const
```

Comment:
What about returning a `const reference` here (and below)? pybind11 will create a copy later on anyway, right?

Reply:
Good idea, even though it's going to be inlined anyways...